src/components/Dashboard.tsx

```diff
@@ -1,46 +1,95 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react'
+import logo from '../../public/images/logo.PNG'
 import Image from 'next/image';
-import { UserData, ComponentProps } from '@/types/types';
-import { getTodaysDate } from '@/utils';
+import { getTodaysDate, getZodiacSign } from '@/utils';
 import Navbar from '../components/Navbar';
 import { insights } from '@/mockdata';
 import Router from "next/router";
-import Logo from '@/components/logo';
+import { PassageUserInfo } from '@passageidentity/passage-elements/passage-user';
+import { AuthProps } from '@/types/types';
+import dotenv from 'dotenv';
+import { Passage } from '@passageidentity/passage-js';
 
+const getCurrentUserInfo = async () => {
+  dotenv.config();
+  const passageAppId = process.env.NEXT_PUBLIC_PASSAGE_APP_ID;
 
-type DashboardProps = ComponentProps & {
-  logOut: () => void
-};
+  if (!passageAppId) {
+    throw new Error('NEXT_PUBLIC_PASSAGE_APP_ID is not defined in the environment.');
+  }
 
+  const passage = new Passage(passageAppId);
+  const user = passage.getCurrentUser();
+  const userInfo = await user.userInfo();
+  return userInfo
+}
 
-export default function Dashboard({ isAuthorized, data, logOut}: DashboardProps) {
-  const [user, setUser] = useState<UserData | null>(null)
-  const [userInsights, setUserInsights] = useState(insights)
-
+export default function Dashboard({ isAuthorized, userID }: AuthProps) {
+  const [user, setUser] = useState<PassageUserInfo | undefined>(undefined);
+  const [demo, setDemo] = useState(false)
+  // const [userInsights, setUserInsights] = useState(insights)
 
   useEffect(() => {
     if (!isAuthorized) {
       Router.push("/");
-    }      
-  
-    if (data) setUser(data[0])
-    //api call needs to be made to get horoscope
-  }, [isAuthorized]);
-  
+    } 
+    else if (userID !== 'ABrrCENR3M0I6XZ7NLA7gNCY') {
+      getCurrentUserInfo()
+        .then((data) => {
+          setUser(data)
+          checkForUser(userID)
+        })
+        .catch((error) => {
+          console.error('Error fetching user:', error);
```

Comment:
should we set an error state here? Maybe something like "We couldn't verify your info"?

Reply:
Passage has an error come up for the user if they can't find them in the database already so I wasn't sure when this would ever happen 😅 if it were to happen I think it would be a server issue. I can add a note saying the server is having issues and try again later?

Reply:
Yeah I think that's a good idea so we have something there even if it's really just an edge case! (: